src/main/kotlin/com/coxautodev/graphql/tools/MethodFieldResolver.kt

```diff
@@ -120,21 +121,43 @@ internal class MethodFieldResolver(field: FieldDefinition, search: FieldResolver
     override fun toString() = "MethodFieldResolver{method=$method}"
 }
 
-open class MethodFieldResolverDataFetcher(private val sourceResolver: SourceResolver, method: Method, private val args: List<ArgumentPlaceholder>): DataFetcher<Any> {
+open class MethodFieldResolverDataFetcher(private val sourceResolver: SourceResolver, method: Method, private val args: List<ArgumentPlaceholder>, private val options: SchemaParserOptions): DataFetcher<Any> {
 
     // Convert to reflactasm reflection
     private val methodAccess = MethodAccess.get(method.declaringClass)!!
     private val methodIndex = methodAccess.getIndex(method.name, *method.parameterTypes)
 
+    private class CompareGenericWrappers {
+        companion object: Comparator<GenericWrapper> {
+            override fun compare(w1: GenericWrapper, w2: GenericWrapper): Int = when {
+                w1.type.isAssignableFrom(w2.type) -> 1
+                else -> -1
+            }
+        }
+    }
```

Comment:
Above basically sorts classes so that most specific ones come first. For example, if you have a bag containing `Object`, `Integer`, `Number`, then ordered result will be `Integer`, `Number`, `Object`.